transcom/mymove
# MAIN B-20808

[INT PR](https://github.com/transcom/mymove/pull/13994)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20808)

## Summary

We want to let the GSR office users to be able to add an appeal to a violation. A violation can only have one appeal by a GSR user and it's a one time thing and there's no requirement for them to edit it or anything. Future work will allow COR office users to add additional appeals and that's why the architecture of the appeals process is an array of objects. This PR does the following:
- adds and edits swagger definitions to allow `gsrAppeals` to appear on report violations and evaluation reports
- adds service object to add a row to the `gsr_appeals` table
- updates UI to allow a GSR user to add an appeal
- adds/updates relevant tests
- added tests to increase server test coverage
- added new `gsr` playwright test

### How to test

1. Turn the GSR flag on in your `envrc`
2. Fire up your `server` and `client`
3. You can either sign in as a multi role user or a GSR user
4. Search for a move (my fav is `EVLRPT` with devseed data)
5. Create a report and add a couple violations
6. Once you're back on the evaluation report screen, click to view the report
7. You should see a `Leave Appeal Decision` button
8. Fill that form out and submit
10. You should now see a dropdown that you can click that will expand and show the details of the appeal you just left underneath the violation
11. Verify you see the information in the `gsr_appeals` table with all relevant data


## Screenshots
![Screenshot 2024-10-23 at 3 45 11 PM](https://github.com/user-attachments/assets/8c478832-b332-48b9-9a27-e27ea671d381)
![Screenshot 2024-10-23 at 3 45 28 PM](https://github.com/user-attachments/assets/d4fecd65-00ea-4d76-ae20-fbc2da48e515)
